pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewSchema.java

```diff
@@ -0,0 +1,176 @@
+   */
+  public static JDBCType fhirTypeToDb(String fhirType) {
+    if (fhirType == null) {
+      return JDBCType.NULL;
+    }
+    switch (fhirType) {
+      case "boolean":
+        return JDBCType.BOOLEAN;
+      case "integer":
+      case "unsignedInt":
+        return JDBCType.INTEGER;
+      case "integer64":
+        return JDBCType.BIGINT;
+      case "decimal":
+        return JDBCType.DOUBLE;
+      case "date":
+        return JDBCType.DATE;
+      case "dateTime":
+      case "instant":
+        return JDBCType.TIMESTAMP;
+      case "time":
+        return JDBCType.TIME;
+      case "base64Binary":
+      case "canonical":
+      case "code":
+      case "id":
+      case "markdown":
+      case "oid":
+      case "string":
+      case "uri":
+      case "url":
+      case "uuid":
+        return JDBCType.VARCHAR;
+    }
+    return null;
+  }
+
+  /**
+   * Creates a DB schema map for a given view.
+   *
+   * @param view the input view
+   * @return an ordered map from column names to DB types
+   */
+  public static ImmutableMap<String, JDBCType> getDbSchema(ViewDefinition view) {
+    ImmutableMap.Builder<String, JDBCType> builder = ImmutableMap.builder();
+    for (Entry<String, Column> entry : view.getColumnTypes().entrySet()) {
+      // This is internally guaranteed.
+      Preconditions.checkState(entry.getValue() != null);
+      JDBCType dbType = fhirTypeToDb(entry.getValue().getType());
+      if (dbType != JDBCType.NULL) {
+        builder.put(entry.getKey(), dbType);
+      } else {
+        log.warn(
+            "No DB type mapping for column {} with type {}; using string instead.",
+            entry.getKey(),
+            entry.getValue());
+        if (Strings.nullToEmpty(entry.getValue().getType()).isEmpty()) {
+          // TODO once we fix column types derivation, this case should be changed to UNDEFINED.
+          builder.put(entry.getKey(), JDBCType.VARCHAR);
+          // builder.put(entry.getKey(), DbType.UNDEFINED);
+        } else {
+          // We may need special handling here for structures but for now use string.
+          builder.put(entry.getKey(), JDBCType.VARCHAR);
+        }
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * Set the values of a given row into the `statement` based on the column information.
+   *
+   * @param rowElements the input elements of the row
+   * @param statement the statement on which `set*()` methods are called to write the row elements
+   * @throws SQLException
+   */
+  public static void setValueInStatement(
+      ImmutableList<RowElement> rowElements, PreparedStatement statement) throws SQLException {
+    int ind = 0;
+    for (RowElement re : rowElements) {
+      if (re.getPrimitive() != null) {
+        // TODO add unit-tests for all cases and add extra cases too if needed!
+        if (ViewApplicator.ID_TYPE.equals(re.getColumnInfo().getInferredType())) {
+          statement.setString(++ind, re.getSingleIdPart());
```

Comment:
For simplicity, can we increment the `ind` at the start of the `for` loop and use it.